arlo_server/models.py

```diff
@@ -357,14 +358,38 @@ class BallotInterpretation(BaseModel):
     __table_args__ = (db.PrimaryKeyConstraint("ballot_id", "contest_id"),)
 
     interpretation = db.Column(db.Enum(Interpretation), nullable=False)
-    # If interpretation is VOTE, contest_choice_id holds the id for the choice
-    # that was voted for. Otherwise null.
-    contest_choice_id = db.Column(
-        db.String(200), db.ForeignKey("contest_choice.id", ondelete="cascade")
-    )
     comment = db.Column(db.Text)
 
 
+# If an ballot is interpretated as VOTE or OVERVOTE, this table stores the
+# choices that the audit board sees are selected on the ballot.
+ballot_interpretation_contest_choice = db.Table(
+    "ballot_interpretation_contest_choice",
+    db.Column(
+        "ballot_id",
+        db.String(200),
+        db.ForeignKey(
+            "ballot_interpretation.ballot_id", primary_key=True, ondelete="cascade"
+        ),
+        nullable=False,
+    ),
+    db.Column(
+        "contest_id",
+        db.String(200),
+        db.ForeignKey(
+            "ballot_interpretation.contest_id", primary_key=True, ondelete="cascade"
+        ),
+        nullable=False,
+    ),
+    db.Column(
+        "contest_choice_id",
+        db.String(200),
+        db.ForeignKey("contest_choice.id", primary_key=True, ondelete="cascade"),
+        nullable=False,
+    ),
```

Comment:
I would add a composite key reference to `(ContestChoice.id, ContestChoice.contest_id)` to make sure this can't contain garbage data.